grace.go

```diff
@@ -369,13 +378,31 @@ type graceServer struct {
 	StopTimeout time.Duration
 }
 
-// start starts the graceServer.
+// start starts the graceServer. If the server's address is a unix socket, it
+// will be removed once the server exits.
 func (gs graceServer) start(ctx context.Context) error {
 	gs.Logger.InfoContext(ctx, "server listening",
 		"server", gs.Name,
 		"address", gs.HTTPServer.Addr,
 	)
-	if err := gs.HTTPServer.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+
+	addr := gs.HTTPServer.Addr
+	network := tcp
+	if strings.HasPrefix(addr, unixPrefix) {
+		addr = strings.TrimPrefix(addr, unixPrefix)
+		network = unix
+	}
+
+	lc := &net.ListenConfig{}
+	l, err := lc.Listen(ctx, network, addr)
```

Comment:
The thing that makes me most uncomfortable with this MR is that we're overloading `addr` to have a special prefix that means "listen over Unix" rather than a dedicated configuration. The way the stdlib gets away without that is by having that dedicated param named `network`.

What do you think about adding a `grace.WithNetwork` option? Then the "default" network for servers can be `"tcp"`, and the functional option can be used to specify `"unix"` or any of the other stdlib-supported network schemes.

---

This also implies that `WaitForTCP` and `WaitForUnix` are actually convenience wrappers around a more generic function:

```go
func WithWaitFor(network, address string) WaitOption
```

Where maybe we deprecate the wrappers and instead just expose the most generic version of the function, for a more flexible interface:

```go
err := grace.Wait(ctx, 10*time.Second, grace.WithWaitFor("tcp", "example.com:80"))
```